src/field/gf_101.rs

```diff
@@ -83,6 +83,8 @@ impl FiniteField for GF101 {
   fn neg_one() -> Self { Self::new(Self::ORDER - 1) }
 
   fn generator() -> Self { Self::new(2) }
+
+  fn from_canonical_u32(n: u32) -> Self { Self::new(n) } // TODO: recheck
```

Comment:
This is a nit, but we don't really need this if we just use `GF101::new()` right?

Reply:
but then that's struct specific function right? We need a trait specific function to create new values, for example: in extension fields, each subvalue in extension field element is base field element.

https://github.com/pluto/ronkathon/blob/9c0ba7055b0f80b0d157243ca48bf002ed823ca4/src/field/gf_101_2.rs#L69

Is there a better way in Rust to do this? Sorry, still in my learning path for Rust.